src/util/integerfactoring.c

```diff
@@ -0,0 +1,610 @@
+			fl->f[i] = f;
+			fl->p[i] = p;
+			return 1;
+		}else if(fl->f[i] == 0){
+			/* This is the biggest factor so far, and a slot still remains. */
+			fl->f[i] = f;
+			fl->p[i] = p;
+			return 1;
+		}
+	}
+	
+	return 0;
+}
+
+int      gaIFLGetFactorPower(GA_FACTOR_LIST* fl, uint64_t f){
+	int i;
+	
+	for(i=0;i<15;i++){
+		if(fl->f[i] == f){
+			return fl->p[i];
+		}
+	}
+	
+	return 0;
+}
+
+uint64_t gaIFLGetProduct(const GA_FACTOR_LIST* fl){
+	uint64_t p = 1;
+	int i, j;
+	
+	for(i=0;i<15;i++){
+		for(j=0;j<fl->p[i];j++){
+			p *= fl->f[i];
+		}
+	}
+	
+	return p;
+}
+
+uint64_t gaIFLGetGreatestFactor(const GA_FACTOR_LIST* fl){
+	uint64_t f = 1;
+	int i;
+	
+	for(i=0;i<15;i++){
+		if(f < fl->f[i]){
+			f = fl->f[i];
+		}
+	}
+	
+	return f;
+}
+
+int      gaIFLsnprintf(char* str, size_t size, const GA_FACTOR_LIST* fl){
+	int    i, j;
+	
+	int    total = 0;
+	size_t left = size;
+	char*  ptr  = size ? str : NULL;
+	
+	/* Loop over all factors and spit them out. */
+	for(i=0;i<15;i++){
+		for(j=0;j<fl->p[i];j++){
+			total += snprintf(ptr, left, "%llu*", (unsigned long long)fl->f[i]);
+			if(ptr){
+				left  -= strlen(ptr);
+				ptr   += strlen(ptr);
+			}
+		}
+	}
+	
+	/* If no factors were printed, print 1. */
+	if(total == 0){
+		total += snprintf(ptr, left, "1*");
+		if(ptr){
+			left  -= strlen(ptr);
+			ptr   += strlen(ptr);
+		}
+	}
+	
+	/* Terminate buffer ('*' -> '\0') and deduct one character. */
+	total--;
+	if(str && size > 0){
+		if(total >= size){
+			str[size-1] = '\0';
+		}else{
+			str[total]  = '\0';
+		}
+	}
+	
+	return total;
+}
+
+
+#if 0
+void runTest(uint64_t n, uint64_t maxN, uint64_t k){
+	char buf[128];
+	GA_FACTOR_LIST fl;
+	
+	int isPrime = gaIIsPrime(n);
+	printf("%llu %s prime.\n", (unsigned long long)n, isPrime ? "is" : "is not");
+	
+	if(k==0 || k>=n || maxN==0 || maxN==n){
+		printf("Attempting exact factorization of %llu.\n", (unsigned long long)n);
+	}
+	if(k>0 && k<n){
+		printf("Factorization will fail if no integer in the range [%llu, %llu] is"
+		       " %llu-smooth.\n",
+		       (unsigned long long)n,
+		       (unsigned long long)(maxN == 0 ? n : maxN),
+		       (unsigned long long)k);
+	}
+	
+	int factorized = gaIFactorize(n, maxN, k, &fl);
+	if(factorized){
+		gaIFLsnprintf(buf, sizeof(buf), &fl);
+		printf("%llu is %llu-smooth and factorizes as %s\n\n",
+		       (unsigned long long)gaIFLGetProduct(&fl),
+		       (unsigned long long)gaIFLGetGreatestFactor(&fl),
+		       buf);
+	}else{
+		printf("Factorization failed.\n\n");
+	}
+}
+
+int main(int argc,char* argv[]){
+	runTest(18446744073709551615ULL,                            0,  4096);
+	runTest(18446744073709551615ULL,                            0,     0);
+	runTest(18446744073709551615ULL,                           -1,     0);
+	runTest( 2196095973992233039ULL,                            0,     0);
+	runTest( 2196095973992233039ULL,  2196095973992233039ULL*1.01,    64);
+	
+	return 0;
+}
+#endif
```

Comment:
Yes, that is a problem.  I haven't figured out a way to go around this, but maybe forcing all the tests to build against the static version would do.

If you figure it out that would be great, but don't spend too much time on this.